tests/ui/infinite/type-param-default-with-diverging-assoc-type.rs

```diff
@@ -0,0 +1,17 @@
+//@ revisions: current next
+//@[next] compile-flags: -Znext-solver
+//@ ignore-compare-mode-next-solver (explicit revisions)
+
+trait Trait {
+    type Diverges<D: Trait>;
+}
+
+impl<T> Trait for T {
+    type Diverges<D: Trait> = D::Diverges<D>;
+}
+
+struct Bar<T: ?Sized = <u8 as Trait>::Diverges<u8>>(Box<T>);
```

Comment:
As there are no where-bounds on `T`, we never normalized the parameter default with the old solver. This is a breaking change